backend/app/core/ipfs.py

```diff
@@ -0,0 +1,32 @@
+import ipfshttpclient
+
+from ipaddress import ip_address, ip_interface, IPv4Address, IPv6Address
+
+from.config import IPFS_ADDRESS, IPFS_PORT
+
+
+class IPFSClient:
+
+    def __init__(self, address: str, port_number: int):
+        
+        assertion_error = f'port number must be 1 <= port number <= 65535 porvided: {port_number}'
+        assert 1 <= port_number <= 65535, assertion_error
+        self.port_number: int = port_number
+        
+        # decide whether given address uses ip4, ip6 or dns4 protocol
+        try:
+            ip_addr: IPv4Address | IPv6Address = ip_address(address)
```

Comment:
Good idea to have union types as it has been introduced in the recent python 3.10 version?

Reply:
@amecklenborg what is the problem here? what would you like to see? I ask because:
- The python version used in the container is 3.9.7 (see the `backend/Dockerfile`) 
-  there is no union type here. Just a type hint that say that variable `ip_addr` is either of type `IPv4Address` or `IPv6Address`

Reply:
I just say this for avoiding posible compatibility issues. Union types are now allowed in [Python3.10](https://www.python.org/downloads/release/python-3100/)

Reply:
So you are saying we should update the python version to 3.10 ?

Reply:
What do you mean with compatibility issues? can you provide an example?

Reply:
yes, I am just saying we can update it to 3.10

Reply:
But what is the point of updating? Just the union type? the there is a new type in 3.10 doesn't mean that the current syntax won't work in 3.10. If this is just the only reason to update the version is a bit overkill. By changing the version it can happen that others things stop working. I think it is just too risky to change the whole setup version just for a typing class.